### PR TITLE
ON-47414 # stop supporting autocomplete for date, datetime and time elements

### PR DESCRIPTION
**Pull request by divporter**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

### CODE REVIEW

**Review by jdawg093 — APPROVED** [review]

File: src/hooks/form-date-picker/useFormDatePickerProps.tsx [hunk 1]
```diff
@@ -64,72 +45,77 @@ export default function useFormDatePickerProps({
...
+              },
+              '&:hover fieldset': {
+                borderColor: '#b5b5b5',
+              },
+              '&.Mui-focused fieldset': {
+                borderColor: '#485fc7',
+                borderWidth: '1px',
+              },
+            },
+          },
```
Comment (mymattcarroll) [inline src/hooks/form-date-picker/useFormDatePickerProps.tsx:91]: Would be nice to get these into CSS and use bulma variables for maintainability

File: src/hooks/form-date-picker/useFormDatePickerProps.tsx [hunk 2]
```diff
@@ -64,72 +45,62 @@ export default function useFormDatePickerProps({
...
+    }),
     [
       ariaDescribedby,
-      autocompleteAttributes,
       className,
       id,
+      maxDateMemo,
+      minDateMemo,
       onBlur,
+      onChange,
```
Comment (mymattcarroll) [inline src/hooks/form-date-picker/useFormDatePickerProps.tsx:98]: We are passing in new `onChange` function every render to `useFormDatePickerProps()` so no point in memo'ing this. I dont think we gain lot anyway.

File: src/styles/date.scss [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    &:hover fieldset {
+      border-color: $input-hover-border-color;
+    }
+    &.Mui-focused fieldset {
+      border-color: $input-focus-border-color;
+      border-width: $control-border-width;
+    }
+  }
+}
```
Comment (mymattcarroll) [inline src/styles/date.scss:18]: ❤️

File: src/hooks/form-date-picker/useFormDatePickerProps.tsx (hunk 2 above)
Comment (divporter) [reply src/hooks/form-date-picker/useFormDatePickerProps.tsx:98]: mmmm yeah I did see that and thought about going up another level.
